kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/FormServiceBase.java

```diff
@@ -0,0 +1,139 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.services.jbpm;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.drools.core.util.StringUtils;
+import org.jbpm.kie.services.impl.form.FormProvider;
+import org.jbpm.services.api.DefinitionService;
+import org.jbpm.services.api.DeploymentService;
+import org.jbpm.services.api.RuntimeDataService;
+import org.jbpm.services.api.UserTaskService;
+import org.jbpm.services.api.model.DeployedUnit;
+import org.jbpm.services.api.model.ProcessDefinition;
+import org.jbpm.services.task.impl.TaskContentRegistry;
+import org.kie.api.task.model.Task;
+import org.kie.internal.runtime.manager.InternalRuntimeManager;
+import org.kie.internal.task.api.ContentMarshallerContext;
+import org.kie.server.services.api.KieServerRegistry;
+import org.kie.server.services.impl.marshal.MarshallerHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FormServiceBase {
+
+    public static final Logger logger = LoggerFactory.getLogger(FormServiceBase.class);
+
+    private DefinitionService definitionService;
+    private DeploymentService deploymentService;
+    private RuntimeDataService dataService;
+    private UserTaskService userTaskService;
+
+    private MarshallerHelper marshallerHelper;
+
+    private Set<FormProvider> providers;
+
+    public FormServiceBase(DefinitionService definitionService, DeploymentService deploymentService,
+            RuntimeDataService dataService, UserTaskService userTaskService,
+            KieServerRegistry context) {
+        this.definitionService = definitionService;
+        this.deploymentService = deploymentService;
+        this.dataService = dataService;
+        this.userTaskService = userTaskService;
+        this.marshallerHelper = new MarshallerHelper(context);
+    }
+
+    public void setProviders(Set<FormProvider> providers) {
+        this.providers = providers;
+    }
+
+    public String getFormDisplayProcess(String containerId, String processId, String lang) {
+        ProcessDefinition processDesc = definitionService.getProcessDefinition(containerId, processId);
+        if (processDesc == null) {
+            throw new IllegalStateException("Process definition " + containerId + " : " + processId + " not found");
+        }
+        Map<String, String> processData = definitionService.getProcessVariables(containerId, processId);
+
+        if (processData == null) {
+            processData = new HashMap<String, String>();
+        }
+
+        Map<String, Object> renderContext = new HashMap<String, Object>();
+        renderContext.put("process", processDesc);
+        renderContext.put("outputs", processData);
+        renderContext.put("marshallerContext", marshallerHelper);
```

Comment:
I'd remove the marshaller helper from here and include it in the provider implementation that comes from jbpm code base